examples/platform/linux/AppMain.cpp

```diff
@@ -120,6 +124,16 @@ int ChipLinuxAppInit(int argc, char ** argv)
     if (LinuxDeviceOptions::GetInstance().mWiFi)
     {
         chip::DeviceLayer::ConnectivityMgrImpl().StartWiFiManagement();
+        while (chip::DeviceLayer::ConnectivityMgrImpl().IsWiFiManagementStarted() == false)
```

Comment:
How about extracting this in a separate method? It seems easeier to read:

```cpp

bool EnsureWifiIsStarted() {
   for (int cnt = 0; cnt < kWifiStartCheckAttempts; cnt++) 
   {
       if (chip::DeviceLayer::ConnectivityMgrImpl().IsWiFiManagementStarted()) {
         return true;
       }

       usleep(kWifiStartCheckTime);
   }

   return chip::DeviceLayer::ConnectivityMgrImpl().IsWiFiManagementStarted();
}

// .....
if (!EnsureWifiIsStarted()) {
   ChipLogError(NotSpecified, "Wi-Fi Management taking too long to start.");
}
```

This would avoid the 'break' and decouples message from check logic.

Is there some comment or info we could add to make the message more actionable? Like explaining that this may trigger the device to go into unprovisioned mode? 

Since this checks for WPA_INTERFACE_CONNECTED, does this mean that any unprovisioned device must have a 500ms (or whatever the config delay is) before it can enter into provisioning mode? I think we should document this near the configuration settings to not have people set it very high with the logic "wait as long as required, even on slow systems".

Reply:
> Since this checks for WPA_INTERFACE_CONNECTED, does this mean that any unprovisioned device must have a 500ms (or whatever the config delay is) before it can enter into provisioning mode? I think we should document this near the configuration settings to not have people set it very high with the logic "wait as long as required, even on slow systems".

WPA_INTERFACE_CONNECTED state signals that the device is able to check status of network interface. The device polls every kWifiStartCheckTimeMs to check WPA_INTERFACE_CONNECTED status has been reached. Only really slow devices may fail kWifiStartCheckAttempts. All devices will wait at least kWifiStartCheckTimeMs during initialization and at most kWifiStartCheckTimeMs * kWifiStartCheckAttempts. We can tweak kWifiStartCheckTimeMs to a lower value and kWifiStartCheckAttempts to a higher value to privilege fast devices.